strapi/strapi
# fix: possible query ambiguity during Strapi v5 db migration

### What does it do?

Add the column's table in the GROUP BY of the queries in the migration 5.0.0-02-created-document-id.

### Why is it needed?

When runner Strapi v5 for the first time after an update from v4, you can have an error during th database migration.

In the migration 5.0.0-02-created-document-id, the GROUP BY in the SQL request does not precise the column's table. This trigger an error when a content type use internationalization and one of its field is named "[singularName_of_the_type]Id".

For example, if in your Strapi v4 you have a content type with:

- singular name "test",
- internationalization set on
- a field named "testId"

When you'll run Strapi v5 for the first time, you will have this error:

```
┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                                                           │
│   MigrationError: Migration 5.0.0-02-created-document-id (up) failed: Original error:                                                     │
│       SELECT `tests`.id as id, group_concat(DISTINCT `inv_test_id`) as other_ids                                                          │
│       FROM `tests`                                                                                                                        │
│       LEFT JOIN `tests_localizations_links` ON `tests`.id = `tests_localizations_links`.`test_id`                                         │
│       WHERE document_id IS NULL                                                                                                           │
│       GROUP BY `test_id`                                                                                                                  │
│       LIMIT 1;                                                                                                                            │
│        - ambiguous column name: test_id                                                                                                   │
│       at /home/mottet/git/strapi-failing-v5-db-migration/node_modules/umzug/lib/umzug.js:169:27                                           │
│       at async Umzug.runCommand (/home/mottet/git/strapi-failing-v5-db-migration/node_modules/umzug/lib/umzug.js:125:20)                  │
│       at async Object.up (/home/mottet/git/strapi-failing-v5-db-migration/node_modules/@strapi/database/dist/index.js:6531:7)             │
│       at async Object.up (/home/mottet/git/strapi-failing-v5-db-migration/node_modules/@strapi/database/dist/index.js:6555:11)            │
│       at async Object.sync (/home/mottet/git/strapi-failing-v5-db-migration/node_modules/@strapi/database/dist/index.js:2246:9)           │
│       at async Strapi.bootstrap (/home/mottet/git/strapi-failing-v5-db-migration/node_modules/@strapi/core/dist/Strapi.js:351:5)          │
│       at async Strapi.load (/home/mottet/git/strapi-failing-v5-db-migration/node_modules/@strapi/core/dist/Strapi.js:315:5)               │
│       at async Module.develop (/home/mottet/git/strapi-failing-v5-db-migration/node_modules/@strapi/strapi/dist/node/develop.js:177:28)   │
│       at async action (/home/mottet/git/strapi-failing-v5-db-migration/node_modules/@strapi/strapi/dist/cli/commands/develop.js:18:5)     │
│       at async Command.parseAsync (/home/mottet/git/strapi-failing-v5-db-migration/node_modules/commander/lib/command.js:923:5)           │
│                                                                                                                                           │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

By adding the column's table in the GROUP BY, we have no more risk of ambiguity.

### How to test it?

You can reproduce this bug with those steps:

1) Create a Strapi v4 project.
2) Create a content type with "test" as singularName and internationalization turned on.
3) Update Strapi to v5.
4) Run Strapi. It will show you the error shared above.

I created a repository with this failing setup: <https://github.com/mottet/strapi-failing-v5-db-migration>

Simply clone it and run those commands to see the issue:

``` bash
yarn install
yarn develop
```

Then you can checkout the branch `use-fix-from-pr` that contains a patch that use this PR fix and run again the commands:

``` bash
yarn install
yarn develop
```

This time, you should have no more issues.
